vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java

```diff
@@ -22,24 +31,64 @@
     @Autowired
     private ISpaceManager spaceManager;
 
+    @Autowired
+    private ISpaceDisplayManager spaceDisplayManager;
+
+    @Autowired
+    private AuthenticationFacade authFacade;
+
     @After("execution(public * edu.asu.diging.vspace.web..*Controller.*(..))")
     public void setExhibition(JoinPoint jp) {
         Object[] args = jp.getArgs();
         MethodSignature signature = (MethodSignature) jp.getSignature();
         Class<?> returnType = signature.getReturnType();
         // only if there is a model object injected and if we're returning a string
         // (assuming that returning a string implies rendering of a view afterwards
-        if (args != null && returnType == String.class) {
-            for (Object obj : args) {
-                if (obj instanceof Model && !(obj instanceof RedirectAttributes)) {
-                    if (!((Model) obj).containsAttribute("exhibition")) {
+        if(args!=null && returnType==String.class) {
+            for(Object obj : args) {
+                if(obj instanceof Model && !(obj instanceof RedirectAttributes)) {
+                    if(!((Model) obj).containsAttribute("exhibition")) {
                         ((Model) obj).addAttribute("exhibition", exhibitionManager.getStartExhibition());
                     }
-                    if (!((Model) obj).containsAttribute("allSpaces")) {
+                    if(!((Model) obj).containsAttribute("allSpaces")) {
                         ((Model) obj).addAttribute("allSpaces", spaceManager.getAllSpaces());
                     }
                 }
             }
         }
     }
+
+    @Around("execution(public * edu.asu.diging.vspace.web.publicview..*Controller.*(..))")
+    public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
+        Object[] args = jp.getArgs();
+        MethodSignature signature = (MethodSignature) jp.getSignature();
+        int indexofModel = (Arrays.asList(signature.getParameterTypes())).indexOf(Model.class);
+        int indexofSpaceId = (Arrays.asList(signature.getParameterTypes())).indexOf(String.class);
+        Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition();
+        ISpace space = spaceManager.getSpace(args[indexofSpaceId].toString());
+        if(exhibition==null) {
+            return "home";
+        }
+        if(exhibition.getMode().equals(ExhibitionModes.ACTIVE)) {
```

Comment:
this line throws a null pointer when there is no mode yet (which is true for all existing exhibitions)